### PR TITLE
GitHub Actions - CI Updates

### LINKED ISSUE #369: GitHub Actions Updates and Extended Python Versions
**Is your feature request related to a problem? Please describe.**
The GitHub Actions pipeline could use a bit of cleaning up which includes updating actions, simplifying names, and building Pure Python Wheels on the latest Python Version. In addition, we should deprecate Python 3.8 support.

**Describe the solution you'd like**
Make adjustments to the `.github` workflow files, and the `pyproject.toml` files.

**Describe alternatives you've considered**
N/A.

**Additional context**
N/A.

### PR DESCRIPTION
**Pull request by srivarra**
**If you haven't already, please read through our [contributing guidelines](https://ark-analysis.readthedocs.io/en/latest/_rtd/contributing.html) before opening your PR**

**What is the purpose of this PR?**

Closes #369.

**How did you implement your changes**

Used the workflows present in angelolab/alpineer. Also added package badges which state the tooling, PyPI info, supported Python Versions, etc...

**Remaining issues**

N/A.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -244,7 +255,7 @@ The median pulse height (MPH) provides a way to assess the sensitivity of the de
 2. `cd` into `toffy`.
 3. Install the pre-commit hooks with `pre-commit install --install-hooks`
 4. Set up Poetry for `toffy`
-   1. Run `poetry install` to install `alpineer` into your virtual environment (this will create a local virtual environment).
+   1. Run `poetry install` to install `toffy` into your virtual environment (this will create a local virtual environment).
```
Comment (camisowers) [inline README.md:258]: Can we add the Developer Notes link to the table of contents?

**Review by camisowers — CHANGES_REQUESTED** [review]
Looks good! Just a minor readme request.

**Review by alex-l-kong — APPROVED** [review]
Looks good

File: README.md [hunk 2]
```diff
@@ -20,6 +32,9 @@ This repo is currently in beta testing. None of the code has been published yet,
 - [Directory structure](#directory-structure)
 - [Panel format](#panel-format)
 - [Median Pulse Height](#median-pulse-height)
+- [Development Notes](#development-notes)
+  - [Requirements](#requirements)
+- [Setup](#setup)
```
Comment (camisowers) [inline README.md:37]: This should probably be the same as requirements, a subheading under development notes right? If we want users to mainly use the conda environment probably best to make it clear Setup is only for developers.

**Review by ngreenwald — APPROVED** [review]
Looks good, agree with Cami

**Review by camisowers — APPROVED** [review]
Stunning.

**Pull request merged**